File: native_src/ios/src/AdMobInterstitial.m
//
//  AdMob ANE Extension
//  iOS Native Extension
//
//  AdMobInterstitial.m
//  AdMobAne
//
//  Copyright (c) 2011-2015 Code Alchemy Inc. All rights reserved.
//

#import "FlashRuntimeExtensions.h"
#import "GADInterstitial.h"
#import "GADInterstitialDelegate.h"
#import "AdMobAne.h"
#import "AdMobInterstitial.h"
#import "AdMobEvents.h"

@implementation AdMobInterstitial
const NSString *INTER_ID = @"InterstitialAd";
@synthesize context;
@synthesize adMobRequest;
@synthesize adMobId;
- (void)interstitialDidReceiveAd:(GADInterstitial *)interstitial
{
    mIsInterstitialLoaded = YES;
    FREDispatchStatusEventAsync(self.context, (UTF8String)INTERSTITIAL_LOADED, (uint8_t*)[INTER_ID UTF8String]);
}
- (void)interstitial:(GADInterstitial *)interstitial didFailToReceiveAdWithError:(GADRequestError *)error;
{
    Log(@"Event didFailToReceiveAdWithError");
    Log([NSString stringWithFormat:@"Error Details: %@",error]);
    FREDispatchStatusEventAsync(self.context, (uint8_t*)INTERSTITIAL_FAILED_TO_LOAD, (uint8_t*)[INTER_ID UTF8String]);
    clearInterstitial();
}

+ (void)interstitialWillPresentScreen:(GADInterstitial *)interstitial;
{
    Log(@"Event interstitialWillPresentScreen");
    FREDispatchStatusEventAsync(self.context, (uint8_t*)INTERSTITIAL_AD_OPENED, (uint8_t*)[INTER_ID UTF8String]);
}
- (void)interstitialDidDismissScreen:(GADInterstitial *)interstitial;
{
    Log(@"Event interstitialDidDismissScreen");
    FREDispatchStatusEventAsync(self.context, (uint8_t*)INTERSTITIAL_AD_CLOSED, (uint8_t*)[INTER_ID UTF8String]);
    clearInterstitial();
}

- (void)interstitialWillLeaveApplication:(GADInterstitial *)interstitial;
{
    FREDispatchStatusEventAsync(self.context, (uint8_t*)INTERSTITIAL_LEFT_APPLICATION, (uint8_t*)[INTER_ID UTF8String]);
}
- (void) clearInterstitial
{
    Log(@"clearInterstitial");
    self.view = nil;
    mInterstitialView = nil;
    context = NULL;
    adMobRequest = NULL;
    adMobId = NULL;
    clearInterstitialInstance();
}

- (void) buildInterstitial
{
    mInterstitialView = [[GADInterstitial alloc] init];
    mInterstitialView.adUnitID = adMobId;
    mIsInterstitialLoaded = NO;
    Log(@"isLoaded set to: NO");
    [mInterstitialView loadRequest:adMobRequest];
    Log(@"Creation Completed");
}
- (void) create
{
    if(mInterstitialView) return;
    id delegate = [[UIApplication sharedApplication] delegate];
    UIWindow * win = [delegate window];
}

- (void) remove
{
    Log(@"remove");
    if(mIsInterstitialLoaded)
    {
        [self clearInterstitial];
    }
}
- (BOOL) isInterstitialLoaded
{
    Log(@"isInterstitialLoaded");
}

- (void) show
{
    Log(@"Show Interstitial");
    [mInterstitialView presentFromRootViewController:self];
}
- (void)viewDidLoad
{
    Log(@"interstitial viewDidLoad");
    [self buildInterstitial];
}
- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}
- (void)viewDidUnload
{
    Log(@"interstitial viewDidUnload");
    mInterstitialView = nil;
    mInterstitialView.delegate = nil;
    context = NULL;
    adMobRequest = NULL;
    adMobId = NULL;
}

File: native_src/android/src/com/codealchemy/ane/admobane/AdMobInterstitial.java
//
//  AdMobAne ANE Extension
//  Android Native Extension
//
//  Copyright (c) 2011-2015 Code Alchemy Inc. All rights reserved.
//

package com.codealchemy.ane.admobane;

import android.app.Activity;
import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.InterstitialAd;

public class AdMobInterstitial {
	private static final String CLASS = "AdMobInterstitial - ";
	public static final String INTER_ID	= "InterstitialAd";
	private ExtensionContext mContext;
	private Activity mActivity;
	private InterstitialAd mInterstitialAd;
    private String mInterstitialId;
	public ExtensionContext getContext() { return mContext; }
	public Activity getActivity() { return mActivity; }
    public AdMobInterstitial(ExtensionContext ctx, Activity act, String interstitialId)
    {
    	mContext		= ctx;
    	mInterstitialId	= interstitialId;
    }
    
	public void create() {
    	mContext.log(CLASS+"create interstitial");
		mInterstitialAd = new InterstitialAd(mActivity);
		mInterstitialAd.setAdUnitId(mInterstitialId);
		AdRequest request = mContext.getRequest();
    	mContext.log(CLASS+"load interstitial Request");
		mInterstitialAd.loadAd(request);
	}
	
	public void remove() {
    	mContext.log(CLASS+"remove");
    	mActivity		= null;
    	mInterstitialId	= null;
	}
	public void cache() {
    	mContext.log(CLASS+"cache interstitial");
    	mContext.log(CLASS+"set interstitial Request");
		AdRequest request = mContext.getRequest();
    	mContext.log(CLASS+"load interstitial Request");
		mInterstitialAd.loadAd(request);
	}
	public Boolean isLoaded() {
    	mContext.log(CLASS+"isLoaded: "+mInterstitialAd.isLoaded());
		return mInterstitialAd.isLoaded();
	}
	public void show() {
    	mContext.log(CLASS+"show");
		mInterstitialAd.show();
	}
}
